extensions/smallrye-metrics/runtime/src/main/java/io/quarkus/smallrye/metrics/runtime/SmallRyeMetricsRecorder.java

```diff
@@ -126,6 +133,25 @@ public void registerMetrics(BeanInfo beanInfo, MemberInfo memberInfo) {
                 memberInfo);
     }
 
+    public void registerMetricFromProducer(String beanId, MetricType metricType,
+            String metricName, String[] tags, String description,
+            String displayName, String unit) {
+        InjectableBean<Object> injectableBean = Arc.container().bean(beanId);
+        BeanManager beanManager = CDI.current().getBeanManager();
```

Comment:
Since you're already using the ArC API you could use `io.quarkus.arc.ArcContainer.beanManager()` instead...